pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,223 @@
+# coding: utf-8
+# Copyright (c) Max-Planck-Institut für Eisenforschung GmbH - Computational Materials Design (CM) Department
+# Distributed under the terms of "New BSD License", see the LICENSE file.
+
+from __future__ import print_function
+
+import numpy as np
+import scipy.constants
+from pyiron.atomistics.master.parallel import AtomisticParallelMaster
+from pyiron_base import JobGenerator
+from sklearn.linear_model import LinearRegression
+from collections import defaultdict
+
+__author__ = "Joerg Neugebauer, Jan Janssen"
+__copyright__ = (
+    "Copyright 2020, Max-Planck-Institut für Eisenforschung GmbH - "
+    "Computational Materials Design (CM) Department"
+)
+__version__ = "1.0"
+__maintainer__ = "Jan Janssen"
+__email__ = "janssen@mpie.de"
+__status__ = "production"
+__date__ = "Sep 1, 2017"
+
+
+eV_div_A3_to_GPa = (
+    1e21 / scipy.constants.physical_constants["joule-electron volt relationship"][0]
+)
+
+def calc_elastic_tensor(strain, stress=None, energy=None, volume=None, rotations=None):
+    """
+    Calculate 6x6-elastic tensor from the strain and stress or strain and energy+volume.
+
+    Rotations matrices can be added to take box symmetries into account (unit matrix can
+    be added but does not have to be included in the list)
+
+    Args:
+        strain (numpy.ndarray): nx3x3 strain tensors
+        stress (numpy.ndarray): nx3x3 stress tensors
+        energy (numpy.ndarray): n energy values
+        volume (numpy.ndarray): n volume values
+        rotations (numpy.ndarray): mx3x3 rotation matrices
+    """
+    if len(strain)==0:
+        raise ValueError('Not enough points')
+    if rotations is not None:
+        rotations = np.append(np.eye(3), rotations).reshape(-1, 3, 3)
```

Comment:
My concern is that the user gives an empty list, which in principle shouldn't happen as long as phonopy is used, but I guess it's a bit more robust. I'm easily overruled if my explanation is not sufficient though :)

Reply:
Right, fair enough. Look at this again though, I guess these two functions should be made private since they're not intended to be used outside elastic.py. In this case I'm less worried about a user passing empty rotations because it would be a developer who would have to do it rather than a user, but even so your point is very reasonable.

Reply:
Ah! That's actually exactly the opposite of what I'd like the user to do. The thing is, these fit functions are perfect and I'd probably also use them just as they are. The problem is the data collection - there I'd like the user to set up the simulations themselves and measure strain tensors etc. as well, which is not difficult at all. The mathematically difficult part is the fitting, and for this you can use these functions.